src/components/PopupWithForm.js

```diff
@@ -32,6 +40,18 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  submitButtonLoadingState(isLoading) {
+    
+      if (this._submitButton) {
+        this._submitButton.textContent = isLoading
+          ? this._defaultButtonText
+          : "Saving...";
+      }
+
+  }
+
+
+
```

Comment:
Please, delete `this.close();` from line 59 in `PopupWithForm`


You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well